src/utils/hash.ts

```diff
@@ -32,27 +32,27 @@ async function encryptValue(value: string, hashedValue: string, dataType: string
     return ""
   }
 
-  const hashValueAlreadyExists = !!userHashedValues[dataType] && userHashedValues[dataType].includes(hashedValue);
 
```

Comment:
seems like we keep going back and forth. my proposal would be like java. this is an important area and it pays to have readable and testable code. 

String hashedValue = produceHash(value);
                boolean isCached = CacheManager.hasCachedValue(dataType, hashedValue);
                if (!isCached) {
                    // doesn't exist yet, encrypt it then update the cache
                    encryptValue(value, hashedValue, dataType);

                    // populate it with the single value
                    CacheManager.addCacheValue(dataType, hashedValue);

                    // populate the hashed values cache
                    populateHashValues();
                }


here are the functions in the cache manager

/**
     * Check to see if a hash value is found for a given data type
     * @param dataType (i.e. file_name)
     * @param matchingHash
     * @return boolean
     */
    public static boolean hasCachedValue(String dataType, String matchingHash) {
        List<String> hashValues = hashMap.get(dataType);
        if (hashValues != null) {
            for (String val : hashValues) {
                if (val.equalsIgnoreCase(matchingHash)) {
                    return true;
                }
            }
        }
        return false;
    }

    public static void updateCacheValues(String dataType, List<String> hashValues) {
        hashMap.put(dataType, hashValues);
    }

    public static void addCacheValue(String dataType, String hashValue) {
        List<String> hashValues = hashMap.get(dataType);
        if (hashValues == null) {
            hashValues = new ArrayList<>();
        }
        hashValues.add(hashValue);
        hashMap.put(dataType, hashValues);
    }

Reply:
Java doesn't have an event loop.   this code does not work in this environment